### PR TITLE
enable `EBITENGINE_INTERNAL_IMAGES_KEY` on browsers

### LINKED ISSUE #2270: enable `EBITENGINE_INTERNAL_IMAGES_KEY` on browsers
This is forked from #2265. Probably downlonding a zip file is fine.

### PR DESCRIPTION
**Pull request by superloach**
Will fix #2270
(I also finally learned how to rebase :tada:)

### CODE REVIEW

File: internal/restorable/images_js.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 The Ebitengine Authors
```
Comment (hajimehoshi) [inline internal/restorable/images_js.go:1]: 2022

**Review by hajimehoshi — COMMENTED** [review]
I'll take a look further later, thanks!

**Review by hajimehoshi — COMMENTED** [review]
Also please update doc.go about `EBITENGINE_INTERNAL_IMAGES_KEY`

File: internal/restorable/images_js.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	a := global.Get("document").Call("createElement", "a")
+	blob := global.Get("Blob").New(
+		[]interface{}{jsData},
+		map[string]interface{}{"type": "archive/zip"},
+	)
+	a.Set("href", global.Get("URL").Call("createObjectURL", blob))
+	a.Set("download", dir+".zip")
+	a.Call("click")
+
+	return nil
```
Comment (hajimehoshi) [inline internal/restorable/images_js.go:60]: Can we move this (and notjs's) implementation of DumpImages to `graphicscommand`? `graphicscommand.DumpImages` would accept `[]*graphicscommand.Image`.

1. We don't have to expose `DumpName`
2. We can reuse the hack to download something from browser by `<a>`.

Comment (superloach) [reply internal/restorable/images_js.go:60]: fantastic idea 👍 I'll do that

Comment (superloach) [reply internal/restorable/images_js.go:60]: How should I invoke `DumpImages` from atlas? Should I pull images from `theBackends`?

Comment (superloach) [reply internal/restorable/images_js.go:60]: Hmm, `theBackends` wouldn't work, since we don't have access to `(*restorable.Image).image` from atlas

Comment (hajimehoshi) [reply internal/restorable/images_js.go:60]: `restorable` already knows the images to dump so you don't have to modify `atlas`, right?

Comment (superloach) [reply internal/restorable/images_js.go:60]: oh gotcha, my bad. my brain isn't awake yet haha

File: internal/graphicscommand/image_js.go [hunk 3]
```diff
@@ -15,43 +15,67 @@
...
-func (i *Image) Dump(graphicsDriver graphicsdriver.Graphics, path string, blackbg bool, rect image.Rectangle) error {
-	// Screen image cannot be dumped.
-	if i.screen {
-		return nil
-	}
-
-	path = strings.ReplaceAll(path, "*", strconv.Itoa(i.id))
-	path = filepath.Base(path)
-
+func jsDownload(buf *bytes.Buffer, mime string, path string) {
```
Comment (hajimehoshi) [inline internal/graphicscommand/image_js.go:26]: `download` is fine

**Review by hajimehoshi — COMMENTED** [review]
Don't forget to update `doc.go` :-)

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

**Comment by superloach** [comment]
Hmm, windows build failed with `panic: directx: ID3D12Device::CreateCommittedResource failed: HRESULT(2147942414) [recovered]`
I've gotten this error a few times before on windows, what is it?

**Comment by hajimehoshi** [comment]
> Hmm, windows build failed with panic: directx: ID3D12Device::CreateCommittedResource failed: HRESULT(2147942414) [recovered]
> I've gotten this error a few times before on windows, what is it?

This is OOM. Your PR is not to blame. Let's run the tests again.

**Comment by superloach** [comment]
Quick google search looks like it's an OOM error?

**Comment by superloach** [comment]
Gotcha. Any way the OOM could be avoided? I've had games crash randomly while developing from my windows machine

**Comment by hajimehoshi** [comment]
> Gotcha. Any way the OOM could be avoided? I've had games crash randomly while developing from my windows machine

It's pretty hard to prevent OOM 100%, but the tests tend to create too many images without a loop, so I think this is a very extreme case.

**Comment by hajimehoshi** [comment]
Oops this seems my mistake. Let me see...

**Comment by hajimehoshi** [comment]
Could you rebase your PR? I think I can merge this tomorrow. Thanks,

**Comment by superloach** [comment]
I think I did that right :sweat_smile: still getting the hang of rebase

**Comment by superloach** [comment]
I think I know what I did wrong... I pulled instead of force pushing. It works out though.

**Pull request merged**